test/middleware/spec.check-progress.js

```diff
@@ -1,242 +1,124 @@
-                req.sessionModel.toJSON().should.contain.all.keys(
-                    'step-1-field-1',
-                    'step-1-field-2',
-                    'step-2-field-1',
-                    'step-2-field-2',
-                    'step-3-field-1',
-                    'step-3-field-2',
-                    'step-4-field-1',
-                    'step-4-field-2',
-                    'fork-field-1',
-                    'fork-field-2'
-                );
-            });
+    describe('checkJourneyProgress', () => {
+        it('calls callback with no arguments if step has no prereqs', () => {
+            controller.options.route = '/one';
+            controller.checkJourneyProgress(req, res, next);
+            next.should.have.been.calledOnce;
+            next.should.have.been.calledWithExactly();
         });
 
-        describe('when a step is skipped', function () {
-
-            beforeEach(function () {
-                req.method = 'POST';
-                Controller.prototype.getForkTarget.returns('/step2');
-                controller.emit('complete', req, res);
-            });
-
-            it('invalidates skipped step', function () {
-                req.sessionModel.get('steps').should.not.contain('/fork1');
-            });
-
-            it('invalidates fields associated with skipped step', function () {
-                expect(req.sessionModel.get('fork-field-1')).to.be.undefined;
-                expect(req.sessionModel.get('fork-field-2')).to.be.undefined;
-            });
-
-            it('doens\'t invalidate the \'next\' step in the current journey', function () {
-                req.sessionModel.get('steps').should.contain('/step2');
-            });
+        it('calls callback with no arguments if prerequisite steps are complete', () => {
+            req.sessionModel.set('steps', [ '/one', '/two' ]);
+            controller.options.route = '/three';
+            controller.checkJourneyProgress(req, res, next);
+            next.should.have.been.calledOnce;
+            next.should.have.been.calledWithExactly();
+        });
 
-            it('doesn\'t unset fields associated with the next step in the current journey', function () {
-                req.sessionModel.get('step-2-field-1').should.be.true;
-                req.sessionModel.get('step-2-field-2').should.be.true;
-            });
+        it('calls callback with no arguments if specified prereqs is in steps', () => {
+            req.sessionModel.set('steps', [ '/one', '/two' ]);
+            controller.options.route = '/four';
+            controller.options.prereqs = [ '/two' ];
+            controller.checkJourneyProgress(req, res, next);
+            next.should.have.been.calledOnce;
+            next.should.have.been.calledWithExactly();
+        });
 
+        it('calls callback with MISSING_PREREQ error code if accessing step that has not had prerequisite steps complete', () => {
```

Comment:
"calls callback with MISSING_PREREQ error for steps with no valid prerequisite"
:+1: